deps/rabbit/src/rabbit_networking.erl

```diff
@@ -562,16 +577,36 @@ handshake(Ref, ProxyProtocolEnabled, BufferStrategy) ->
                 {error, protocol_error, Error} ->
                     failed_to_recv_proxy_header(Ref, Error);
                 {ok, ProxyInfo} ->
-                    {ok, Sock} = ranch:handshake(Ref),
+                    Sock = try_ranch_handshake(Ref),
                     ok = tune_buffer_size(Sock, BufferStrategy),
                     {ok, {rabbit_proxy_socket, Sock, ProxyInfo}}
             end;
         false ->
-            {ok, Sock} = ranch:handshake(Ref),
+            Sock = try_ranch_handshake(Ref),
             ok = tune_buffer_size(Sock, BufferStrategy),
             {ok, Sock}
     end.
 
+try_ranch_handshake(Ref) ->
+    try ranch:handshake(Ref) of
+        {ok, Sock} ->
+            Sock
```

Comment:
There's no reason no to return `{ok, Sock}`. You could also use the name `ranch_handshake` as the "try" doesn't tell us much.